python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -35,16 +46,186 @@ def __init__(self, params, lr):
             self._rank = 0
             self._world_size = 1
 
+    def local_state_dict(self):
+        """Return the state pertaining to current rank of the optimizer.
```

Comment:
BTW, could we add an example of what the dict looks like to let devs have a clear idea on what are saved.

Reply:
added